### PR TITLE
z/TPF Clean Build Updates for OMR Part I

### PR DESCRIPTION
**Pull request by jjohnst-us**
This Commit contains several updates related to getting
a clean compilations for the z/TPF platform.

Signed-off-by: jjohnst-us <jjohnst@us.ibm.com>

### CODE REVIEW

**Comment by jjohnst-us** [comment]
Here is a successful IBM internal build link:
http://vmfarm.ottawa.ibm.com/build_info.php?build_id=359108

This is a low risk pull request from the perspective of other platforms, as all changes were to get a clean compilation.

**Comment by mstoodle** [comment]
@jjohnst-us please do not post IBM internal URLs into this project. Not everyone in the community can read those links (because not everyone has access to IBM servers), so they shouldn't be in the community.

Thanks!

**Comment by jjohnst-us** [comment]
@mstoodle 
Apologies, fixed.  Let me know if you need me to do anything else.

**Comment by jjohnst-us** [comment]
@charliegracie 
Hey Charlie, whenever you get a chance to look at this much appreciated.  Low risk changes from other platforms perspective.

File: port/ztpf/omrosdump.c [hunk 1]
```diff
@@ -254,24 +254,24 @@ omrdump_create(struct OMRPortLibrary *portLibrary, char *filename, char *dumpTyp
...
 			}
 			// +2 for terminating Null and joining path seperator
 			oldPath = alloca(
-					strlen(targetDirectory) + strlen(s->argv.wkSpace) + 2);
+					strlen(targetDirectory) + strlen((const char *)s->argv.wkSpace) + 2);
 			strcpy(oldPath, targetDirectory);
 			strcat(oldPath, "/");
-			strcat(oldPath, s->argv.wkSpace);
-			s->argv.rc = rename(oldPath, JVMfn); /* Success. Return. */
+			strcat(oldPath, (const char *)s->argv.wkSpace);
```
Comment (charliegracie) [inline port/ztpf/omrosdump.c]: Is all of the casting between `const char*` and `char *` required?

Comment (jjohnst-us) [inline port/ztpf/omrosdump.c]: If I remove it I get this compilation error:
./ztpf/omrosdump.c: In function 'omrdump_create':
./ztpf/omrosdump.c:273:4: error: pointer targets in passing argument 2 of 'strcat' differ in signedness [-Werror=pointer-sign]
/ztpf/commit/opensource/include/string.h:142:21: note: expected 'const char * __restrict__' but argument is of type 'uint8_t *'

That particular structure member is in lowcore for us...  with that said we probably could have organized the types and their conversion better.  If it was a mainline platform path I would definitely do something about it, but this is z/TPF specific.

**Pull request merged**